openmc/model/surface_composite.py

```diff
@@ -50,6 +52,155 @@ def __pos__(self):
+        self._axis = axis
+        c1, c2 = center
+
+        # Coords for axis-perpendicular planes
+        ctop = c1 + r1
+        cbottom = c1 - r1
+
+        cright = c2 + r1
+        cleft = c2 - r1
+
+        # Side lengths
+        if r2 > r1 * sqrt(2):
+            warnings.warn(f'r2 is greater than sqrt(2) * r1. Octagon may' + \
+                          ' be erroneous.')
+        if r1 > r2 * sqrt(2):
+            warnings.warn(f'r1 is greater than sqrt(2) * r2. Octagon may' + \
+                          ' be erroneous.')
+
+        L_perp_ax1 = (r2 * sqrt(2) - r1) * 2
+        L_perp_ax2 = (r1 * sqrt(2) - r2) * 2
+
+        # Coords for quadrant planes
+        p1_ur = array([L_perp_ax1/2, r1, 0.])
+        p2_ur = array([r1, L_perp_ax1/2, 0.])
+        p3_ur = array([r1, L_perp_ax1/2, 1.])
+
+        p1_lr = array([r1, -L_perp_ax1/2, 0.])
+        p2_lr = array([L_perp_ax1/2, -r1, 0.])
+        p3_lr = array([L_perp_ax1/2, -r1, 1.])
+
+        points = [p1_ur, p2_ur, p3_ur, p1_lr, p2_lr, p3_lr]
+
+        # Orientation specific variables
+        if axis == 'z':
+            coord_map = [0,1,2]
+            self.top = openmc.YPlane(y0=ctop, **kwargs)
+            self.bottom = openmc.YPlane(y0=cbottom, **kwargs)
+            self.right = openmc.XPlane(x0=cright, **kwargs)
+            self.left = openmc.XPlane(x0=cleft, **kwargs)
+        elif axis == 'y':
+            coord_map = [0,2,1]
+            self.top = openmc.ZPlane(z0=ctop, **kwargs)
+            self.bottom = openmc.ZPlane(z0=cbottom, **kwargs)
+            self.right = openmc.XPlane(x0=cright, **kwargs)
+            self.left = openmc.XPlane(x0=cleft, **kwargs)
+        elif axis == 'x':
+            coord_map = [2,0,1]
+            self.top = openmc.ZPlane(z0=ctop, **kwargs)
+            self.bottom = openmc.ZPlane(z0=cbottom, **kwargs)
+            self.right = openmc.YPlane(y0=cright, **kwargs)
+            self.left = openmc.YPlane(y0=cleft, **kwargs)
+
+        # Put our coordinates in (x,y,z) order
+        calibrated_points = []
+        for p in points:
+            calibrated_points += [p[coord_map]]
+
+        p1_ur, p2_ur, p3_ur, p1_lr, p2_lr, p3_lr = calibrated_points
+
+        self.upper_right = openmc.Plane.from_points(p1_ur, p2_ur, p3_ur,
+                                                    **kwargs)
+        self.lower_right = openmc.Plane.from_points(p1_lr, p2_lr, p3_lr,
+                                                    **kwargs)
+        self.lower_left = openmc.Plane.from_points(-p1_ur, -p2_ur, -p3_ur,
+                                                   **kwargs)
+        self.upper_left = openmc.Plane.from_points(-p1_lr, -p2_lr, -p3_lr,
+                                                   **kwargs)
+
+
+    def __neg__(self):
+        region1 = -self.top & +self.bottom & -self.right & +self.left
+        if self._axis == 'y':
+            region2 = -self.upper_right & -self.lower_right & \
+                +self.lower_left & +self.upper_left
```

Comment:
If you were to change `coord_map` for the y-axis case to `[1, 2, 0]`, would that prevent us having to use the `if` block here?